cyclonedx/model/component.py

```diff
@@ -26,9 +26,181 @@
+            )
+
+        self.uid = uid
+        self.url = url
+        self.author = author
+        self.committer = committer
+        self.message = message
+
+    @property
+    def uid(self) -> Optional[str]:
+        """
+        A unique identifier of the commit. This may be version control specific. For example, Subversion uses revision
+        numbers whereas git uses commit hashes.
+
+        Returns:
+            `str` if set else `None`
+        """
+        return self._uid
+
+    @uid.setter
+    def uid(self, uid: Optional[str]) -> None:
+        self._uid = uid
+
+    @property
+    def url(self) -> Optional[XsUri]:
+        """
+        The URL to the commit. This URL will typically point to a commit in a version control system.
+
+        Returns:
+             `XsUri` if set else `None`
+        """
+        return self._url
+
+    @url.setter
+    def url(self, url: Optional[XsUri]) -> None:
+        self._url = url
+
+    @property
+    def author(self) -> Optional[IdentifiableAction]:
+        """
+        The author who created the changes in the commit.
+
+        Returns:
+            `IdentifiableAction` if set else `None`
+        """
+        return self._author
+
+    @author.setter
+    def author(self, author: Optional[IdentifiableAction]) -> None:
+        self._author = author
+
+    @property
+    def committer(self) -> Optional[IdentifiableAction]:
+        """
+        The person who committed or pushed the commit
+
+        Returns:
+            `IdentifiableAction` if set else `None`
+        """
+        return self._committer
+
+    @committer.setter
+    def committer(self, committer: Optional[IdentifiableAction]) -> None:
+        self._committer = committer
+
+    @property
+    def message(self) -> Optional[str]:
+        """
+        The text description of the contents of the commit.
+
+        Returns:
+            `str` if set else `None`
+        """
+        return self._message
+
+    @message.setter
+    def message(self, message: Optional[str]) -> None:
+        self._message = message
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Commit):
+            return hash(other) == hash(self)
+        return False
+
+    def __hash__(self) -> int:
+        return hash((self.uid, hash(self.url), hash(self.author), hash(self.committer), self.message))
+
+    def __repr__(self) -> str:
+        return f'<Commit uid={self.uid}, url={self.url}, message={self.message}>'
+
+
+class ComponentEvidence:
+    """
+    Our internal representation of the `componentEvidenceType` complex type.
+
+    Provides the ability to document evidence collected through various forms of extraction or analysis.
+
+    .. note::
+        See the CycloneDX Schema definition: https://cyclonedx.org/docs/1.4/xml/#type_componentEvidenceType
+    """
+
+    def __init__(self, licenses: Optional[List[LicenseChoice]] = None,
+                 copyright_: Optional[List[Copyright]] = None) -> None:
```

Comment:
why the underscore in `copytight_`? 
is there another variable called `copyright`? i assume there is a module with that name.

Reply:
`copyright` is a keyword and I was following [PEP-0008](https://www.python.org/dev/peps/pep-0008/#id42)